main.cpp

```diff
@@ -117,25 +120,28 @@ struct MyUpdater
 {
     static float updateLogic(U* that, float* newValue )        //10
     {
-        if (newValue == nullptr)
+        if (newValue == nullptr && newValue == nullptr)
         {
             std::cout<<"Detected Nullpointer!" << std::endl;
-            return 0.0f; 
         }
```

Comment:
you're checking `newValue` twice.  
- [ ] revise

- [ ] add space around your `<<` so it's easier to read.

Reply:
Fixed both Cases